tests/Properties.hs

```diff
@@ -1614,6 +1614,36 @@ prop_read_write_file_D x = unsafePerformIO $ do
         (const $ do y <- D.readFile f
                     return (x==y))
 
+prop_hgetline_like_s8_hgetline (LinedASCII filetext) (lineEndIn, lineEndOut) = idempotentIOProperty $ do
+    tid <- myThreadId
+    let f = "qc-test-"++show tid
+    let newlineMode = NewlineMode (if lineEndIn then LF else CRLF) (if lineEndOut then LF else CRLF)
+    bracket_
+        (writeFile f filetext)
+        (removeFile f)
+        $ do 
+            bsLines <- withFile f ReadMode (\h -> do
+                hSetNewlineMode h newlineMode
+                readByLines C.hGetLine h
+              )
+            sLines <- withFile f ReadMode (\h -> do
+                hSetNewlineMode h newlineMode
+                readByLines System.IO.hGetLine h
+              )
```

Comment:
Would it be possible to reduce code duplication here?

Reply:
One way would be to write a `readFileByLines` function.

```haskell
readFileByLines filename getLine = withFile filename ReadMode (\h -> do
    hSetNewlineMode h newlineMode
    readByLines getLine h
  )
```

Alternatively, because it is expensive to open files on Windows it would be possible to use just the `Handle` from when the file was written and just seek the beginning after the initial write and read.
Something like

```haskell
hPutStr h_ filetext
hFlush h_

let readByLinesFromStart getLine = hSeek h_ AbsoluteSeek 0 *> readByLines getLine h_
   
hSetNewlineMode h_ newlineMode
bsLines <- readByLinesFromStart C.hGetLine
sLines <- readByLinesFromStart System.IO.hGetLine
```

Then instead of open 3 files per test-case, we open 1 file.
I can try running that on Windows later to see if it is a large improvement (I would expect it is close to 3x for this short test).
Downside being that it isn't as obvious as opening files from scratch.

Reply:
I do not particularly care about performance of the test suite, I'd rather keep tests as straightforward and focused as possible.